src/crepe/facade/SDLContext.cpp

```diff
@@ -383,101 +276,124 @@ ivec2 SDLContext::get_size(const Texture & ctx) {
-									  {sdl_to_keycode(event.key.keysym.scancode), false},
-									  {},
-									  {}});
+			}
+
+			case SDL_KEYUP: {
+				this->update_keyboard_state();
+				EventData transfer_event;
+				transfer_event.event_type = SDLContext::EventType::KEYUP;
+				transfer_event.data.key_data = KeyData{
+					.key = sdl_to_keycode(event.key.keysym.scancode),
+					.key_repeat = false,
+				};
+				event_list.push_back(transfer_event);
 				break;
-			case SDL_MOUSEBUTTONDOWN:
-				event_list.push_back({SDLContext::EventType::MOUSEDOWN,
-									  {},
-									  {sdl_to_mousebutton(event.button.button),
-									   {event.button.x, event.button.y}},
-									  {}});
+			}
+
+			case SDL_MOUSEBUTTONDOWN: {
+				EventData transfer_event;
+				transfer_event.event_type = SDLContext::EventType::MOUSEDOWN;
+				transfer_event.data.mouse_data = MouseData{
+					.mouse_button = sdl_to_mousebutton(event.button.button),
+					.mouse_position = mouse_pos,
+				};
+				event_list.push_back(transfer_event);
 				break;
-			case SDL_MOUSEBUTTONUP:
-				event_list.push_back({SDLContext::EventType::MOUSEUP,
-									  {},
-									  {sdl_to_mousebutton(event.button.button),
-									   {event.button.x, event.button.y}},
-									  {}});
+			}
+			case SDL_MOUSEBUTTONUP: {
+				EventData transfer_event;
+				transfer_event.event_type = SDLContext::EventType::MOUSEUP;
+				transfer_event.data.mouse_data = MouseData{
+					.mouse_button = sdl_to_mousebutton(event.button.button),
+					.mouse_position = mouse_pos,
+				};
+				event_list.push_back(transfer_event);
 				break;
-			case SDL_MOUSEMOTION:
-				event_list.push_back({SDLContext::EventType::MOUSEMOVE,
-									  {},
-									  {{},
-									   {event.motion.x, event.motion.y},
-									   -1,
-									   INFINITY,
-									   {event.motion.xrel, event.motion.yrel}},
-									  {}});
+			}
+
+			case SDL_MOUSEMOTION: {
+				EventData transfer_event;
+				transfer_event.event_type = SDLContext::EventType::MOUSEMOVE;
+				transfer_event.data.mouse_data = MouseData{
+					.mouse_position = mouse_pos,
+					.rel_mouse_move = {event.motion.xrel, event.motion.yrel},
+				};
+				event_list.push_back(transfer_event);
 				break;
-			case SDL_MOUSEWHEEL:
-				event_list.push_back(
-					{SDLContext::EventType::MOUSEWHEEL,
-					 {},
-					 {{}, {}, event.wheel.y < 0 ? -1 : 1, event.wheel.preciseY, {}},
-					 {}});
+			}
+
+			case SDL_MOUSEWHEEL: {
+				EventData transfer_event;
+				transfer_event.event_type = SDLContext::EventType::MOUSEWHEEL;
+				transfer_event.data.mouse_data = MouseData{
+					.mouse_position = mouse_pos,
+					.scroll_direction = event.wheel.y < 0 ? -1 : 1,
+					.scroll_delta = event.wheel.preciseY,
+				};
+				event_list.push_back(transfer_event);
 				break;
-
-			// Forward window events for further processing
+			}
 			case SDL_WINDOWEVENT:
-				this->handle_window_event(event.window, event_list);
+				handle_window_event(event.window, event_list);
```

Comment:
`this->` :(

Check het ook even voor alle andere member functies want ik zie het nu vaker